### PR TITLE
Creates the Github Source Documentation

### PR DESCRIPTION
**Pull request by JeffNeff**

### CODE REVIEW

**Comment by JeffNeff** [comment]
relates https://github.com/triggermesh/docs/issues/158

File: docs/sources/github.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Event Source for Github
+
+This event source creates a webhook to listen for incoming Github Events, turning received requests into [CloudEvents][ce] to be consumed by other TriggerMesh components.
+
+## Prerequisite(s)
+
+- GitHub Tokens
```
Comment (cab105) [inline docs/sources/github.md:7]: Could you add some additional details about whether this should be an oAuth token or personal access token.

File: docs/sources/github.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+### Create GitHub Tokens
+Create a personal access token for GitHub that the GitHub source can use to register webhooks with the GitHub API. Also decide on a secret token that your code will use to authenticate the incoming webhooks from GitHub (secretToken).
+
+The token can be named anything you find convenient. The Source requires repo:public_repo and admin:repo_hook, to let it fire events from your public repositories and to create webhooks for those repositories. Copy and save this token; GitHub will force you to generate it again if misplaced.
```
Comment (cab105) [inline docs/sources/github.md:12]: How about annotating the oauth permissions `repo:public_repo` and `admin:repo_hook`? Also, are there any additional permissions required for private repos as well?

File: docs/sources/github.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+In the Source creation form, give a name to the event source and add the following information:
+
+- **Secret**: Reference to a [TriggerMesh secret][tm-secret] containing an Access Token and Secret Token, as described in the previous sections.
+- **Name**: all TriggerMesh components need a unique name per namespace.
+- **Broker**: request converted into [CloudEvents][ce] will be sent to this location.
+- **Repository owner and Name**: A valid GitHub public repository owned by your GitHub user. (eg. <YOUR USER>/<YOUR REPO>).
+- **Event Types**: Select from the dropdown the types of events the source should emit. 
+
+After clicking the `Save` button, you will be taken back to the Bridge editor. Proceed to adding the remaining
```
Comment (cab105) [inline docs/sources/github.md:30]: How about, `Proceed by adding the remaining...`

File: docs/sources/github.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+components to the Bridge, then submit it.
+
+A ready status on the main _Bridges_ page indicates that the event source is ready to receive notifications from the Github Event Source
```
Comment (cab105) [inline docs/sources/github.md:33]: Missing `.`

File: docs/sources/github.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+![Bridge status](../images/bridge-status-green.png)
+### Verify
+
+Verify the GitHub webhook was created by looking at the list of webhooks under the Settings tab in your GitHub repository. A hook should be listed that points to your Knative cluster with a green check mark to the left of the hook URL, as shown below.
+
```
Comment (cab105) [inline docs/sources/github.md:39]: Where's the image?

File: docs/sources/github.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+### More Information 
+More information on the Github Event Source can be found here -> https://knative.dev/docs/eventing/samples/github-source/
```
Comment (cab105) [inline docs/sources/github.md:41]: How about using a `:` instead of an `->`?

File: docs/sources/github.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+## Event Types
+
+The Github event source emits events that begin with `dev.knative.source.github.` and end in the event type. For example `dev.knative.source.github.pull_request`, `dev.knative.source.github.create`, and `dev.knative.source.github.delete`.
```
Comment (cab105) [inline docs/sources/github.md:45]: Missing a `:` after example.

File: docs/sources/github.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+[tm-secret]: ../guides/secrets.md
```
Comment (cab105) [inline docs/sources/github.md:47]: Missing reference to `[ce]`.

**Review by cab105 — CHANGES_REQUESTED** [review]
A few minor, and a couple not so minor nits.

File: docs/sources/github.md (hunk 1 above)
Comment (JeffNeff) [reply docs/sources/github.md:7]: In the block below it is elaborated.

File: docs/sources/github.md (hunk 2 above)
Comment (JeffNeff) [reply docs/sources/github.md:12]: I dont. know. I pulled this directly from here -> https://knative.dev/docs/eventing/samples/github-source/

**Review by cab105 — APPROVED** [review]

**Pull request merged**